eng/performance/benchmark_jobs.yml

```diff
@@ -65,28 +67,19 @@ jobs:
         strategy:
           matrix:
             ${{ each framework in parameters.frameworks }}:
-              coreclr_${{ framework }}:
-                _Category: coreclr
-                _Framework: ${{ framework }}
-                _CompilationMode: NoTiering
-                _BuildConfig: ${{ parameters.architecture }}_coreclr_$(_Framework)
-              corefx_${{ framework }}:
-                _Category: corefx
-                _Framework: ${{ framework }}
-                _CompilationMode: NoTiering
-                _BuildConfig: ${{ parameters.architecture }}_corefx_$(_Framework)
-              ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}: # to reduce the number of CI legs for every PR
-                ${{ if and(startsWith(framework, 'netcoreapp'), ne(framework, 'netcoreapp2.0')) }}: # Tiered JIT was introduced in .NET Core 2.1
-                  coreclr_${{ framework }}_tiered:
-                    _Category: coreclr
-                    _Framework: ${{ framework }}
-                    _CompilationMode: Tiered
-                    _BuildConfig: ${{ parameters.architecture }}_coreclr_$(_Framework)_$(_CompilationMode)
-                  corefx_${{ framework }}_tiered:
-                    _Category: corefx
-                    _Framework: ${{ framework }}
-                    _CompilationMode: Tiered
-                    _BuildConfig: ${{ parameters.architecture }}_corefx_$(_Framework)_$(_CompilationMode)
+              ${{ each category in parameters.categories }}:
+                ${{ category }}_${{ framework }}:
+                  _Category: ${{ category }}
+                  _Framework: ${{ framework }}
+                  _CompilationMode: NoTiering
+                  _BuildConfig: ${{ parameters.architecture }}_${{ category }}_$(_Framework)
+                ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}: # to reduce the number of CI legs for every PR
+                  ${{ if and(startsWith(framework, 'netcoreapp'), ne(framework, 'netcoreapp2.0')) }}: # Tiered JIT was introduced in .NET Core 2.1
+                    ${{ category }}_${{ framework }}_tiered:
+                      _Category: ${{ category }}
+                      _Framework: ${{ framework }}
+                      _CompilationMode: Tiered
+                      _BuildConfig: ${{ parameters.architecture }}_${{ category }}_$(_Framework)_$(_CompilationMode)
```

Comment:
we may need to specify kind in the build config.

Reply:
AzureDevOps has complained about not unique job ids, so I added it to the job id. 

https://github.com/dotnet/performance/blob/e1cce9b2336c38e2dd3bdd396707452a4d57eb56/eng/performance/benchmark_jobs.yml#L20

Do you think that I should add it to both places?

Reply:
The BuildConfig is the thing that specifies things for mc.dot.net. When we don't have completely unique BuildConfigs between runs that happened on the same queue, mc overwrites the logs (so you could have a failure in the micro leg but not in the ml leg, but if the ml leg finished 2nd, you won't have the logs for the micro leg). I noticed this when first setting up the jobs.

Reply:
aa, now I see it. I am going to change it and add a comment

Reply:
Done